provider/testdata/data/baked/plurals/cardinal_v1/mod.rs

```diff
@@ -0,0 +1,26 @@
+// @generated
+#![cfg(feature = "icu_plurals")]
+type DataStruct =
+    <::icu_plurals::provider::CardinalV1Marker as ::icu_provider::DataMarker>::Yokeable;
+pub fn lookup(locale: &icu_provider::DataLocale) -> Option<&'static DataStruct> {
+    static KEYS: [&str; 12usize] = [
+        "ar", "bn", "en", "es", "fil", "fr", "ja", "ru", "sr", "th", "tr", "und",
+    ];
+    static DATA: [&DataStruct; 12usize] = [
+        &include!("ar.rs.data"),
+        &include!("bn.rs.data"),
+        &include!("en.rs.data"),
+        &include!("es.rs.data"),
+        &include!("fil.rs.data"),
+        &include!("fr.rs.data"),
+        &include!("ja+th+und.rs.data"),
```

Comment:
suggestion: I think we should make intermediate statics instead of relying on LLVM to optimize here.

Reply:
But LLVM does optimise...

Reply:
It's probably a little bit better to be explicit, but I'm not familiar enough with the LLVM optimizer to know what it is able to automatically optimize. Does it automatically combine two static variables with the exact same bit pattern? If we can rely on it to do that, it seems harmless.

Is there a compile time impact, though? Because rustc sees the code with all the duplicate includes and it has to compile all of them repeatedly.

Reply:
As with all optimizations, it's hard to say it can be _relied_ on especially in situations that might be seen as an edge case, and I think databake is big enough to count.

Just as an example of why that property may not hold; if these end up being handled in separate codegen units they probably won't get coalesced unless LLVM has special handling. Typically you don't have to worry about leaf-y stuff getting split into CGUs, but this crate is big enough that I kinda think all bets are off here. Each of these statics contains references to more and more statics and I can easily see LLVM deciding to bucket the static assemblage weirdly, even though at first glance this mod.rs seems pretty close to being a leaf node.

For codesize stuff like this I think if we have a surefire way to lock down a win, we should do it, even if LLVM seems to be doing it.